client/src/main/java/org/eclipse/hono/client/impl/HonoClientImpl.java

```diff
@@ -66,13 +85,14 @@ public HonoClientImpl(final Vertx vertx, final ConnectionFactory connectionFacto
             this.vertx = Vertx.vertx();
         }
         this.connectionFactory = connectionFactory;
+        this.clientConfigProperties = new ClientConfigProperties();
```

Comment:
this constructor should simply invoke
`this(vertx, connectionFactory, new ClientConfigProperties())`